src/lib/viewers/media/DashViewer.js

```diff
@@ -505,6 +505,23 @@ class DashViewer extends VideoBaseViewer {
                 this.handleDownloadError(error, downloadURL);
                 return;
             }
+
+            if (
+                normalizedShakaError.code === shaka.util.Error.Code.BAD_HTTP_STATUS &&
+                normalizedShakaError.data[1] === 401 // token expired
+            ) {
+                this.options
```

Comment:
I think the retry logic should be in the viewers itself, because the likelihood is that the streaming itself will fail, not the refreshing of the token